src/components/__tests__/query-result.test.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { render, cleanup, screen } from '@testing-library/react';
+import QueryResult from '../query-result';
```

Comment:
That's a weird component, given how you use it. Why the null-check in the children?